packages/web-components/fast-foundation/src/horizontal-scroll/horizontal-scroll.ts

```diff
@@ -0,0 +1,363 @@
+            this.nextFlipper.classList.toggle(
+                "disabled",
+                this.isRtl
+                    ? position - this.width <= lastStop
+                    : position + this.width >= lastStop
+            );
+        }
+    }
+
+    /**
+     * Scrolls items to the left
+     * @public
+     */
+    public scrollToPrevious(): void {
+        const position: number = this.getScrollPosition();
+        const stops: Array<number> = this.scrollStops;
+        if (this.isRtl) stops.sort((a: number, b: number): number => b - a);
+        const current: number = stops.findIndex(
+            (stop: number, index: number, array: Array<number>): boolean =>
+                this.isRtl
+                    ? stop <= position
+                    : stop <= position &&
+                      (index === array.length - 1 || array[index + 1] > position)
+        );
+        const right: number = stops[current + 1];
+        const nextIndex: number =
+            stops.findIndex((stop: number): boolean =>
+                this.isRtl ? stop - this.width < right : stop + this.width > right
+            ) || 0;
+        const left: number = this.scrollStops[
+            nextIndex < current ? nextIndex : current > 0 ? current - 1 : 0
+        ];
+        this.scrollToPosition(left, position);
+    }
+
+    /**
+     * Scrolls items to the right
+     * @public
+     */
+    public scrollToNext(): void {
+        const position: number = this.getScrollPosition();
+        const current: number = this.scrollStops.findIndex(
+            (stop: number): boolean => stop === position
+        );
+        const outOfView: number = this.scrollStops.findIndex((stop: number): boolean =>
+            this.isRtl ? stop <= position - this.width : stop >= position + this.width
+        );
+        let nextIndex: number = current;
+        if (outOfView > current + 2) {
+            nextIndex = outOfView - 2;
+        } else if (current < this.scrollStops.length - 2) {
+            nextIndex = current + 1;
+        }
+        const nextStop: number = this.scrollStops[nextIndex];
+        this.scrollToPosition(nextStop, position);
+    }
+
+    /**
+     * Handles scrolling with easing
+     * @param position - starting position
+     * @param newPosition - position to scroll to
+     */
+    private scrollToPosition(
+        newPosition: number,
+        position: number = this.scrollContainer.scrollLeft
+    ): void {
+        const getEasedTime = (ease: ScrollEasing, k: number): number => {
+            if (k > 1) k = 1;
+            switch (ease) {
+                case "ease-in":
+                    return Math.pow(k, 1.675);
+                case "ease-out":
+                    return 1 - Math.pow(1 - k, 1.675);
+                case "ease-in-out":
+                    return 0.5 * (Math.sin((k - 0.5) * Math.PI) + 1);
+                default:
+                    return k;
+            }
+        };
```

Comment:
Can this function be converted into either a static method or extracted to be its own thing?